File: kiddoz_in_flutter/lib/utils/Recommendation.dart
import 'package:kiddoz_in_flutter/utils/TypeEnum.dart';

class Recommendation {
  int id = 0;
  String title = "";
  String description = "";
  String benefits = "";
  int age = 0;
  TypeEnum type = TypeEnum.other;

  Recommendation(this.id, this.title, this.description, this.benefits, this.age, this.type);

  String toStr() {
    return "$id $title $description $benefits $age ${type.name}";
  }
}

File: Recommendation/app/src/main/java/com/example/recommendation/model/Recommendation.kt
package com.example.recommendation.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "recommendation_table")
data class Recommendation(
    @ColumnInfo(name="title") var title: String,
    @ColumnInfo(name="recommendedAge")var recommendedAge: Int,
    @ColumnInfo(name="benefits")var benefits: String,
    @ColumnInfo(name="description")var description: String,
    @ColumnInfo(name="type")var type: TypeEnum,
    @PrimaryKey(autoGenerate = true) @ColumnInfo(name="id") var id: Int?,
    @ColumnInfo(name="sentToServer")var sentToServer: Boolean=true


)

